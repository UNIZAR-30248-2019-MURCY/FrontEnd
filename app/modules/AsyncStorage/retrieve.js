import { AsyncStorage } from 'react-native';

export const retrieveItem = async (key) => {
    let data ='';
    try {
        data = await AsyncStorage.getItem(key)
        console.log(data)
    } catch (error) {
        console.log(error.message);
    }
    return data;
}
