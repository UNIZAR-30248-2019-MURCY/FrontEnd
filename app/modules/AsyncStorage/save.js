import { AsyncStorage } from 'react-native';

export const saveData = async (key, data) => {
    try {
        await AsyncStorage.setItem(key, JSON.stringify(data));
        console.log('data stored');
    } catch (error) {
        // Error retrieving data
        console.log(error.message);
    }
};
