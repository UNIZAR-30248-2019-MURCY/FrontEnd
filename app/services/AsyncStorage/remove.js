import { AsyncStorage } from 'react-native';

export const removeItem = async (key) => {
    try {
        await AsyncStorage.removeItem(key);
        console.log('data removed');
        return true;
    } catch (error) {
        console.log(error.message);
        return false;
    }
}
