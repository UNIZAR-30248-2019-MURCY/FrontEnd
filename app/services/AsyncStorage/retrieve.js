import { AsyncStorage } from 'react-native';

export const retrieveItem = async (key) => {
    let data ='';
    try {
        let dataStored = await AsyncStorage.getItem(key)
        data = JSON.parse(dataStored).token;
        console.log('retrievedItem')
    } catch (error) {
        console.log(error.message);
    }
    return data;
}
