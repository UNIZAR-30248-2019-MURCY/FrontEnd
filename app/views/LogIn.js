import React, { Component } from 'react';
import {
  View,
  TextInput,
  StyleSheet,
  Image,
  Text
} from 'react-native'
import { Button } from 'react-native-elements';
import { Link } from './../config/routing';

export default class LogIn extends Component {
  state = {
    username: '', password: '', email: '', phone_number: ''
  }
  onChangeText = (key, val) => {
    this.setState({ [key]: val })
  }
  logIn = async () => {
    const { email, password } = this.state
    try {
      
      // here place your signup logic
      //console.log('user successfully signed up!: ', success)
    } catch (err) {
      //console.log('error signing up: ', err)
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <Image
          style={styles.logo}
          source={require('./../assets/images/murcy.png')}
        />
        <TextInput
          style={styles.input}
          placeholder='Email'
          autoCapitalize="none"
          placeholderTextColor='darkgrey'
          onChangeText={val => this.onChangeText('email', val)}
        />
        <TextInput
          style={styles.input}
          placeholder='Password'
          secureTextEntry={true}
          autoCapitalize="none"
          placeholderTextColor='darkgrey'
          onChangeText={val => this.onChangeText('password', val)}
        />

        <Button
          buttonStyle={styles.button}
          title="Log In"
          onPress={() => {this.logIn}} />

        <Link to="/signin">
          <Text>Sign Up</Text>
        </Link>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  input: {
    width: 350,
    height: 55,
    backgroundColor: 'whitesmoke',
    margin: 10,
    padding: 8,
    color: 'grey',
    borderRadius: 14,
    fontSize: 18,
    fontWeight: '500',
  },
  container: {
    flex: 1,
    alignItems: 'center',
    //justifyContent: 'center',
    marginTop: 100,
    padding: 20
  },
  logo: {
    margin: 30,
    padding: 8,
    width: 100,
    height: 100
  },
  signUpScreenButton: {
    marginRight: 40,
    marginLeft: 40,
    marginTop: 10,
    paddingTop: 10,
    paddingBottom: 10,
    backgroundColor: 'grey',
    borderRadius: 10,
  },
  button: {
    width: 150,
    height: 55,
    margin: 40,
    padding: 8,
    backgroundColor: 'grey',
    borderRadius: 14
  }

})