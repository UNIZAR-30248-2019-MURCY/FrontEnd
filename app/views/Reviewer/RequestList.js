import React, {Component} from 'react';
import {AsyncStorage, FlatList, StyleSheet, TextInput, View,} from 'react-native'
import {Button, ListItem, Text} from 'react-native-elements';
import {getRequestEdit, requestEdit} from "../../services/user/userFuncs";
import {reviewerReqList} from "../../services/user/reviewerFuncs";

export default class RequestList extends Component {

    constructor(props) {
        super(props);
        this.state = {
            description: '',
            token: '',
            requests: false,
            errorGettingReq: false,
            notShow: true
        }
        this.retrieveItem = this.retrieveItem.bind(this);
    }

    async retrieveItem(key) {
        try {
            console.log('retrieveItem')
            let data = await AsyncStorage.getItem(key)
            let token = JSON.parse(data).token;
            this.setState({token: token})
        } catch (error) {
            console.log(error.message);
        }
    }

    componentDidMount() {
        this.retrieveItem('token')
            .then(r => {
                reviewerReqList(this.state.token)
                    .then((data) => {
                        this.setState({requests: data})
                        this.setState({notShow: false})
                        console.log(this.state.requests)
                    })
                    .catch((error) => {
                        this.setState({errorGettingReq: error.message})
                    })
            })

    }



    render() {
        let showReq = (
            this.state.notShow || this.state.errorGettingReq ?
                <View style={styles.error}>
                    <Text style={{color: 'red'}}>
                        {this.state.errorGettingReq}
                    </Text>
                </View> :
                    <View style={styles.containerRequestExist}>
                        <FlatList
                            data={this.state.requests}
                            keyExtractor={item => item.id.toString()}
                            renderItem={({item}) => (
                                <ListItem
                                    title={item.description}
                                    titleStyle={{fontSize: 18}}
                                    bottomDivider
                                    onPress={() => {
                                        let workflow = item.workflow;

                                        while(workflow.nextWorkflow !== null){
                                            workflow = workflow.nextWorkflow;
                                        }

                                        this.props.navigation.navigate('RequestDetails', {
                                            workflow: workflow
                                        });
                                    }}
                                />
                            )}
                        />
                    </View>
        );

        return (
            <View style={styles.container}>
                <View style={styles.containerTitle}>
                    <Text h2>Requests List</Text>
                </View>
                {showReq}
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 20
        //justifyContent: 'center',
    },
    containerTitle: {
        alignItems: 'center',
        marginTop: 50,
        padding: 20
    },
    containerRequest: {
        flex: 1,
        alignItems: 'center',
        marginTop: 20,
    },
    containerRequestExist: {
        flex: 1,
        marginTop: 20,
    },
    containerRequestExistContent: {
        flex: 1,
        alignItems: 'flex-start',
        marginTop: 20,
        fontSize: 18,
        color: 'grey',
    },
    cross: {
        marginTop: 50,
        marginBottom: 40,
        marginRight: 10,
        alignItems: 'flex-end',
    },

    logo: {
        marginBottom: 30,
        padding: 8,
        width: 100,
        height: 100
    },
    input: {
        width: 300,
        height: 100,
        backgroundColor: 'whitesmoke',
        margin: 10,
        padding: 8,
        color: 'grey',
        borderRadius: 14,
        fontSize: 18,
        fontWeight: '500',
    },
    button: {
        width: 150,
        height: 55,
        marginTop: 20,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    button2: {
        width: 150,
        height: 55,
        borderRadius: 14
    },
    error: {
        margin: 10,
        alignItems: 'center',
    },

})
