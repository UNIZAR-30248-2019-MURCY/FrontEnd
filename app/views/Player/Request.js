import React, {Component} from 'react';
import {ActivityIndicator, AsyncStorage, ScrollView, StyleSheet, TextInput, View,} from 'react-native'
import {Button, Text} from 'react-native-elements';
import {getRequestEdit, requestEditor, editRequestEditor} from "../../services/user/userFuncs";
import {retrieveItem} from "../../modules/AsyncStorage/retrieve";

export default class Request extends Component {

    constructor(props) {
        super(props);
        this.state = {
            description: '',
            token: '',
            request: false,
            workflow: false,
            errorGettingReq: false,
            errorForm: false,
            editing: false,
            reRequest: false,
            loading: false
        }
        this.onChangeText = this.onChangeText.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.searchLastRequest = this.searchLastRequest.bind(this);
    }

    onChangeText = (key, val) => {
        this.setState({[key]: val})
    }

    searchLastRequest() {
        while (this.state.workflow.nextWorkflow !== null) {
            this.setState({request: this.state.workflow.nextWorkflow})
        }
    }

    componentDidMount() {
        this.setState({loading: true})
        retrieveItem('token')
            .then(data => {
                this.setState({token: JSON.parse(data).jsonWebToken})
                getRequestEdit(this.state.token)
                    .then((request) => {
                        if(request)
                        {
                            this.setState({request: request})
                            this.setState({workflow: request.workflow})
                            this.searchLastRequest()
                        }
                        this.setState({loading: false})
                    })
                    .catch((error) => {
                        this.setState({errorGettingReq: error.message}),
                            this.setState({loading: false})
                    })
            })
    }

    handleSubmit() {
        this.setState({loading: true})
        if (this.state.description !== '') {
            if (!this.state.editing) {
                requestEditor(this.state.description, this.state.token)
                    .then((data) => {
                        this.setState({loading: false})
                        this.props.navigation.replace('RequestConfirm');
                    })
                    .catch((error) => {
                        this.setState({errorForm: error.message}),
                        this.setState({loading: false})
                    })
            } else {
                editRequestEditor(this.state.description, this.state.token)
                    .then((data) => {
                        this.setState({loading: false})
                        this.props.navigation.replace('RequestConfirm');
                    })
                    .catch((error) => {
                        this.setState({errorForm: error.message}),
                            this.setState({loading: false})
                    })
            }
        } else {
            this.setState({loading: false})
            this.setState({errorForm: 'Introduzca todos los campos'})
        }

    }

    render() {
        let showErr = (
            this.state.errorForm ?
                <View style={styles.error} className='errorShow'>
                    <Text style={{color: 'red'}}>
                        {this.state.errorForm}
                    </Text>
                </View> :
                <View></View>
        );

        let showLoading = (
            <View style={ styles.horizontal}>
                <ActivityIndicator animating={this.state.loading} size="large" color="grey" />
            </View>
        );

        let reqNotExist = (
            <View style={styles.containerRequest}  className='editCreateReq'>
                <TextInput
                    className='descriptionInput'
                    style={styles.input}
                    multiline={true}
                    placeholder='Description'
                    autoCapitalize="none"
                    placeholderTextColor='darkgrey'
                    onChangeText={val => this.onChangeText('description', val)}
                />

                {showErr}

                <Button
                    className='send-button'
                    buttonStyle={styles.button}
                    title="Send"
                    onPress={() => {
                        this.handleSubmit()
                    }}/>

                <Button
                    className='cancel-button'
                    type="clear"
                    buttonStyle={styles.button2}
                    title="Cancel"
                    titleStyle={{color: 'grey'}}
                    onPress={() => {
                        this.props.navigation.goBack();
                    }
                    }/>
            </View>
        );

        let editReq = (
            !this.state.editing ?
                <View style={styles.containerEdit} className='editShow'>
                    <Button
                        className='edit-button'
                        buttonStyle={styles.buttonEdit}
                        title="Edit request"
                        onPress={() => {
                            this.setState({editing: true})
                        }}/>
                </View>
                : reqNotExist
        );

        let reReq = (
            !this.state.reRequest ?
                <View style={styles.containerEdit} className='reReqShow'>
                    <Button
                        className='reReq-button'
                        buttonStyle={styles.buttonEdit}
                        title="Request Again"
                        onPress={() => {
                            this.setState({reRequest: true})
                        }}/>
                </View>
                : reqNotExist
        );

        let reqExist = (
            this.state.request.closed && !this.state.request.approved ?
                <View className='requesDenied'>
                    <Text h4>A request has been denied</Text>
                    <Text style={styles.containerRequestExistContent}>
                        Description: {this.state.workflow.description}
                        {'\n'}{'\n'}
                        Status: {this.state.workflow.status}
                    </Text>
                    {reReq}
                </View>
                :
                this.state.request.closed && this.state.request.approved ?
                    <View className='requesApproved'>
                        <Text h4>A request has been accepted</Text>
                        <Text style={styles.containerRequestExistContent}>
                            Description: {this.state.workflow.description}
                            {'\n'}{'\n'}
                            Status: {this.state.workflow.status}
                        </Text>
                    </View >
                    : <View style={styles.containerInfo} className='editMode'>
                        <Text h4>A request already exists</Text>
                        <Text style={styles.containerRequestExistContent}>
                            Description: {this.state.workflow.description}
                            {'\n'}{'\n'}
                            Status: {this.state.workflow.status}
                        </Text>
                        {editReq}
                    </View>
        );

        let show = (
            this.state.errorGettingReq ?
                <View style={styles.error} className='errorGettingReq'>
                    <Text style={{color: 'red'}}>
                        {this.state.errorGettingReq}
                    </Text>
                </View> :
                !this.state.loading ?
                    this.state.request ?
                        <View style={styles.containerRequestExist} className='requestShow'>
                            {reqExist}
                                <Button
                                    className='return-button'
                                    type="clear"
                                    buttonStyle={styles.button2}
                                    title="Return"
                                    titleStyle={{color: 'grey'}}
                                    onPress={() => {
                                        this.props.navigation.goBack();
                                    }
                                    }/>
                        </View>
                     :reqNotExist
                : showLoading
        );

        return (
            <View style={styles.container}>
                <View style={styles.containerTitle}>
                    <Text h2>Request</Text>
                </View>
                <ScrollView>
                    {show}
                </ScrollView>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 20
    },
    horizontal: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        padding: 10,
        marginTop: 50,
    },
    containerTitle: {
        alignItems: 'center',
        marginTop: 50,
        padding: 20
    },
    containerRequest: {
        flex: 1,
        alignItems: 'center',
        marginTop: 25,
    },
    containerRequestExist: {
        flex: 1,
        padding: 30,
        alignItems: 'center',
    },
    containerRequestExistContent: {
        textAlign: 'center',
        marginTop: 20,
        fontSize: 18,
        color: 'grey',
    },
    cross: {
        marginTop: 50,
        marginBottom: 40,
        marginRight: 10,
        alignItems: 'flex-end',
    },
    logo: {
        marginBottom: 30,
        padding: 8,
        width: 100,
        height: 100
    },
    input: {
        width: 250,
        height: 90,
        backgroundColor: 'whitesmoke',
        margin: 10,
        padding: 8,
        color: 'grey',
        borderRadius: 14,
        fontSize: 18,
        fontWeight: '500',
    },
    button: {
        width: 150,
        height: 55,
        marginTop: 20,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    button2: {
        width: 150,
        height: 55,
        borderRadius: 14
    },
    error: {
        margin: 10,
        alignItems: 'center',
    },
    containerEdit: {
        marginTop: 30,
        alignItems: 'center',
    },
    buttonEdit: {
        width: 150,
        height: 50,
        marginTop: 20,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    containerInfo: {
        alignItems: 'center',
    },

})
