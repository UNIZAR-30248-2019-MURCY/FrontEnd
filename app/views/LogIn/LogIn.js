import React, {Component} from 'react';
import {
    View,
    TextInput,
    StyleSheet,
    Image,
    Text,
    ActivityIndicator
} from 'react-native'
import {Button} from 'react-native-elements';
import Icon from 'react-native-vector-icons/FontAwesome';
import {logInUser, userInfo} from "../../services/user/userFuncs";
import {saveData} from "../../services/AsyncStorage/save";
import {NavigationActions, StackActions} from "react-navigation";


export default class LogIn extends Component {

    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            token: '',
            error: false,
            userData: '',
            loading: false
        }
        this.onChangeText = this.onChangeText.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    onChangeText = (key, val) => {
        this.setState({[key]: val})
    }

    handleSubmit() {
        if (this.state.username !== '' && this.state.password !== '') {
            this.setState({loading: true})
            this.setState({error: false})
            logInUser(this.state.username, this.state.password)
                .then((data) => {
                    saveData('token', data).then(r =>
                        userInfo(data.jsonWebToken).then((dataUser) => {
                                saveData('role', dataUser.role).then(r =>
                                        console.log(dataUser.role),
                                    this.setState({loading: false}),
                                    this.props.navigation.navigate('App')
                                )
                            }
                        )
                            .catch((error) => {
                                this.setState({error: error.message}),
                                    this.setState({loading: false})
                            })
                    );
                })
                .catch((error) => {
                    this.setState({error: error.message}),
                        this.setState({loading: false})
                });
        } else {
            this.setState({error: 'Introduzca todos los campos'})
        }
    }

    render() {
        let showErr = (
            this.state.error ?
                <View style={styles.error} className='errorShow'>
                    <Text style={{color: 'red'}}>
                        {this.state.error}
                    </Text>
                </View> :
                <View></View>
        );
        let showLoading = (
            this.state.loading ?
                <View style={[styles.containerLoading]}>
                    <ActivityIndicator animating={this.state.loading} size="large" color="grey"/>
                </View> :
                <View></View>
        );

        return (

            <View style={styles.container}>
                <View style={styles.cross}>
                    <Button
                        className='close-button'
                        type="clear"
                        icon={
                            <Icon
                                name="times"
                                size={30}
                                color="grey"
                            />
                        }
                        onPress={() => {
                            this.props.navigation.navigate('AuthLoading')
                        }
                        }
                    />
                </View>

                <View style={styles.login}>
                    <Image
                        style={styles.logo}
                        source={require('../../assets/images/murcy.png')}
                    />
                    <TextInput
                        className='userInput'
                        style={styles.input}
                        placeholder='Username'
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onChangeText={val => this.onChangeText('username', val)}
                    />
                    <TextInput
                        className='passInput'
                        style={styles.input}
                        placeholder='Password'
                        secureTextEntry={true}
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onSubmitEditing={() => {
                            this.handleSubmit()
                        }}
                        onChangeText={val => this.onChangeText('password', val)}
                    />
                    {showErr}
                    {showLoading}
                    <Button
                        className='login-button'
                        buttonStyle={styles.button}
                        title="Log In"
                        onPress={() => {
                            this.handleSubmit()
                        }}/>
                    <Button
                        className='signup-button'
                        type="clear"
                        buttonStyle={styles.button2}
                        title="Sign Up"
                        titleStyle={{color: 'grey'}}
                        onPress={() => {
                            this.props.navigation.replace('SignUp');
                        }
                        }/>
                </View>

            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    containerLoading: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        padding: 10
    },
    cross: {
        marginTop: 50,
        marginBottom: 40,
        marginRight: 10,
        alignItems: 'flex-end',
    },
    login: {
        alignItems: 'center',
    },
    logo: {
        marginBottom: 30,
        padding: 8,
        width: 100,
        height: 100
    },
    input: {
        width: 300,
        height: 55,
        backgroundColor: 'whitesmoke',
        margin: 10,
        padding: 8,
        color: 'grey',
        borderRadius: 14,
        fontSize: 18,
        fontWeight: '500',
    },
    button: {
        width: 150,
        height: 55,
        marginTop: 20,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    button2: {
        width: 150,
        height: 55,
        borderRadius: 14
    },
    error: {
        margin: 10,

    },

})
