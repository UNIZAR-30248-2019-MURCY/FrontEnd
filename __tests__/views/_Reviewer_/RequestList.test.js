import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import RequestList from "../../../app/views/Reviewer/RequestList";
import {retrieveItem} from "../../../app/services/AsyncStorage/retrieve";
import Request from "../../../app/views/Player/Request";
import LogIn from "../../../app/views/LogIn/LogIn";

describe('<RequestList />', () => {
    it('RequestList renders without crashing', async () => {
        await retrieveItem();
        const rendered = renderer.create(<RequestList/>).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('RequestList test against snapshot', async () => {
        await retrieveItem();
        const page = renderer.create(<RequestList/>).toJSON();
        expect(page).toMatchSnapshot();
    });
    it('should render the errorGettingReq if state.errorGettingReq is true', async () => {
        await retrieveItem();
        let wrapper = shallow(<RequestList/>);
        wrapper.setState({errorGettingReq: 'Error'});
        expect(wrapper.find('.errorGettingReq').length).toBe(1);
    });
    it('should render the List if state.notShow is false', async () => {
        await retrieveItem();
        let wrapper = shallow(<RequestList/>);
        wrapper.setState({ errorGettingReq: false });
        wrapper.setState({ loading: false });
        expect(wrapper.find('.reqList').length).toBe(1);
    });
});
