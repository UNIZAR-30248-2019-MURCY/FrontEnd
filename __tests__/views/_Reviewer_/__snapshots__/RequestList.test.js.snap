// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestList /> .renderItemIsSelected should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`<RequestList /> RequestList test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 50,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 42.5,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Editors Requests
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginTop": 50,
        "padding": 10,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="grey"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "elevation": undefined,
            "justifyContent": "flex-end",
            "zIndex": undefined,
          },
        ],
        undefined,
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "elevation": undefined,
          "justifyContent": "flex-end",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": undefined,
        }
      }
    />
    <View
      pointerEvents="box-none"
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "elevation": undefined,
              "justifyContent": "flex-end",
              "zIndex": undefined,
            },
          ],
          Object {
            "alignItems": "flex-end",
          },
          Object {
            "paddingVertical": 30,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginHorizontal": 30,
              "zIndex": undefined,
            },
            false,
            false,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(128, 128, 128, 1)",
                "borderRadius": 28,
                "height": 56,
                "width": 56,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 28,
                  "height": 56,
                  "justifyContent": "center",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                  "width": 56,
                }
              }
            >
              <Text
                allowFontScaling={false}
                className="filter-inactive"
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ï‚°
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
